homeassistant/components/mqtt/mixins.py

```diff
@@ -64,6 +64,8 @@
 CONF_VIA_DEVICE = "via_device"
 CONF_DEPRECATED_VIA_HUB = "via_hub"
 
+DEFAULT_ENTITY_NAME = "MQTT Entity"
```

Comment:
The individual component schemas change this to "MQTT Alarm" etc, with the intention it will take precedence.
Suggest to add a new test helper `help_test_default_name` in tests/components/mqtt/test_common.py + call that test from each component to verify the default name is the overridden one, and not `MQTT Entity`.

```suggestion
# Overridden by component schemas
DEFAULT_ENTITY_NAME = "MQTT Entity"
```

Reply:
This would be a cosmetic test though, not a functional one. Is that ok?

In addition, i can remove the default value, in which case it would default to `DOMAIN PLATFORM` (ie. "mqtt switch").

Reply:
Oh, it does not use `DOMAIN` as the second part, for example `alarm_control_panel` would be just `mqtt àlarm`

Reply:
The way I imagine this could potentially break in the future is if the config is validated by the `MQTTEntity` using the `MQTT_ENTITY_BASE_SCHEMA`.

Hence the suggestion to add functional test `help_test_default_name(hass, mqtt_mock, domain, config, expected_name)` which sets up an entity and tests that the default name is as expected.

Reply:
Yes, dropping `CONF_NAME`, default naming is more sensible.